packages/process/src/node/process-manager.ts

```diff
@@ -4,22 +4,26 @@
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
-import { injectable } from 'inversify';
+import { injectable, inject } from 'inversify';
 import { Process } from './process';
 import { Emitter, Event } from '@theia/core/lib/common';
+import { ILogger } from '@theia/core/lib/common/logger';
 
 @injectable()
 export class ProcessManager {
 
-    protected readonly processes: Map<number, Process> = new Map();
     protected id: number = 0;
-    protected readonly deleteEmitter = new Emitter<number>();
+    protected readonly processes: Map<number, Process>;
+    protected readonly deleteEmitter: Emitter<number>;
+
+    constructor( @inject(ILogger) protected logger: ILogger) {
+        this.processes = new Map();
+        this.deleteEmitter = new Emitter<number>();
+    }
 
     register(process: Process): number {
-        const id = this.id;
-        this.processes.set(id, process);
-        this.id++;
-        return id;
+        this.processes.set(++this.id, process);
+        return this.id;
```

Comment:
I just wanted the id to start at 0 .. you mind if if is -1 then ?

Reply:
I do not understand what you mean. Could you please explain this to me?

Reply:
I mean doing ++this.id from this.id = 0... will make the first process have id 1 instead of 0

Reply:
Got it!

> you mind if if is -1 then

`-1` confused me. I will restore the original behavior. Thanks for the reminder!

Reply:
```
this.processes.set(this.id++, process);
return this.id 
```
This would work actually :)

Reply:
Strangely, changing to `this.id++` breaks the `Terminal Backend Contribution is data received from the terminal ws server` test. I have to check why.